src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Batch.cs

```diff
@@ -670,7 +670,7 @@ internal void StatementCompletedHandler(object sender, StatementCompletedEventAr
                     // Not a user cancellation error, add all 
                     foreach (var error in errors)
                     {
-                        int lineNumber = error.LineNumber + Selection.StartLine;
+                        int lineNumber = error.LineNumber + (Selection != null ? Selection.StartLine : 0);
```

Comment:
>0 [](start = 103, length = 1)

Did you verify in sqlops? Just want to make sure the start line starts from 0 not 1